src/handlers/default.ts

```diff
@@ -191,7 +191,11 @@ export class InngestCommHandler {
 
           const stepRes = await this.runStep(fnId, stepId, req.body);
 
-          return void res.json(stepRes);
+          if (stepRes.status === 500) {
+            return void res.status(stepRes.status).json(stepRes.error);
```

Comment:
What if this is a 4xx?

Reply:
@tonyhb For this and the `2xx` question, atm this is an implementation detail.

The user just returns data, and then we give that data `status: 200` if the step runs successfully, and `status: 500` if it throws. `status: 400` is reserved for the client believing it's being called incorrectly (i.e. `POST` without specifying a `fnId` or `stepId`).

This comes back to whether we expose the `status`/`body` pattern to the user or keep it abstracted.

My view has been that it's still an implementation detail; knowing which status codes produce which behaviour (e.g. `500` = retry) feels like an extra thing for the user to learn. We should discuss this!